### DISCUSSION TITLE
close a handle without close  file descriptor

### DISCUSSION THREAD

**Discussion by calvin2021y** [libuv/libuv#4228] [Q&A]
I need use uv_accept to receive a fd send by pipe, but the fd it self is not a pipe.

so after receive it, I need to remove the temporary pipe object after I get the fd by uv_fileno.    in the same time I need to keep the fd not closed.

`uv_unref(h);` not work here,   after I call `uv_unref`, the fd still in loop, confirmed by `uv_walk`. 

`uv_close` seems close the fd too.

Is there a way to do it ?

**Comment by bnoordhuis** [comment]
Won't work on Windows but on Unices you can simply dup() the file descriptor.

**Reply by calvin2021y** [reply]
thanks for the tips.

**Reply by vtjnash** [reply]
That also works on windows (except for socket objects on certain rare configurations)
